lib/segment/src/index/field_index/mmap_point_to_values.rs

```diff
@@ -0,0 +1,389 @@
+    }
+
+    fn read_from_mmap(bytes: &[u8], offset: usize) -> OperationResult<Self> {
+        Ok(Self {
+            lon: f64::read_from_prefix(&bytes[offset..])
+                .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))?,
+            lat: f64::read_from_prefix(&bytes[offset + std::mem::size_of::<f64>()..])
+                .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))?,
+        })
+    }
+
+    fn write_to_mmap(&self, bytes: &mut [u8], offset: usize) -> OperationResult<()> {
+        self.lon
+            .write_to_prefix(&mut bytes[offset..])
+            .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))?;
+        self.lat
+            .write_to_prefix(&mut bytes[offset + std::mem::size_of::<f64>()..])
+            .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))
+    }
+}
+
+impl<'a> MmapValue<'a> for &'a str {
+    fn mmaped_size(&self) -> usize {
+        self.as_bytes().len() + std::mem::size_of::<u32>()
+    }
+
+    fn read_from_mmap(bytes: &'a [u8], offset: usize) -> OperationResult<&'a str> {
+        let bytes = &bytes[offset..];
+        let size = u32::read_from_prefix(bytes)
+            .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))?
+            as usize;
+
+        let bytes = &bytes[std::mem::size_of::<u32>()..];
+        if bytes.len() < size {
+            return Err(OperationError::service_error(
+                NOT_ENOUGHT_BYTES_ERROR_MESSAGE,
+            ));
+        }
+
+        let bytes = &bytes[..size];
+
+        let string = std::str::from_utf8(bytes).map_err(|_| OperationError::service_error(format!("UTF8 conversion error while reading {POINT_TO_VALUES_PATH}. Is the storage corrupted?")))?;
+        Ok(string)
+    }
+
+    fn write_to_mmap(&self, bytes: &mut [u8], offset: usize) -> OperationResult<()> {
+        let bytes = &mut bytes[offset..];
+        if bytes.len() < self.mmaped_size() {
+            return Err(OperationError::service_error(
+                NOT_ENOUGHT_BYTES_ERROR_MESSAGE,
+            ));
+        }
+
+        u32::write_to_prefix(&(self.len() as u32), bytes)
+            .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))?;
+
+        let bytes = &mut bytes[std::mem::size_of::<u32>()..];
+        self.as_bytes()
+            .iter()
+            .enumerate()
+            .for_each(|(i, &b)| bytes[i] = b);
+        Ok(())
+    }
+}
+
+/// Flattened memmaped points-to-values map
+/// It's an analogue of `Vec<Vec<N>>` but in memmaped file.
+/// This structure doesn't support adding new values, only removing.
+/// It's used in mmap field indices like `MmapMapIndex`, `MmapNumericIndex`, etc to store points-to-values map.
+/// This structure is not generic to avoid boxing lifetimes for `&str` values.
+pub struct MmapPointToValues {
+    file_name: PathBuf,
+    mmap: MmapSlice<u8>,
```

Comment:
It was important because of `flusher()` which should be presented to call flush from flushing thread. But now we decided to use deleted flags and this map doesn't support deletions. Removed deletion and changed to `Mmap`